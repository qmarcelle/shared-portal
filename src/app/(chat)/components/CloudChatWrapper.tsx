'use client';

// CloudChatWrapper injects the Genesys Cloud Messenger SDK and manages Messenger-specific events.
// All script loading, Messenger events, and errors are logged for traceability and debugging.

import { useEffect } from 'react';
import { useChatStore } from '../stores/chatStore';

export default function CloudChatWrapper() {
  const { userData } = useChatStore();

  useEffect(() => {
    // Build chatSettings from env and userData
    window.chatSettings = {
      bootstrapUrl: process.env.NEXT_PUBLIC_GENESYS_BOOTSTRAP_URL || '',
      widgetUrl: process.env.NEXT_PUBLIC_GENESYS_WIDGET_URL || '',
      clickToChatJs: process.env.NEXT_PUBLIC_GENESYS_CLICK_TO_CHAT_JS || '',
      clickToChatEndpoint: process.env.NEXT_PUBLIC_CLICK_TO_CHAT_ENDPOINT || '',
      chatTokenEndpoint: process.env.NEXT_PUBLIC_CHAT_TOKEN_ENDPOINT || '',
      coBrowseEndpoint: process.env.NEXT_PUBLIC_COBROWSE_LICENSE_ENDPOINT || '',
      opsPhone: process.env.NEXT_PUBLIC_OPS_PHONE || '',
      opsPhoneHours: process.env.NEXT_PUBLIC_OPS_HOURS || '',
      ...userData,
    };
    console.debug('[CloudChatWrapper] chatSettings:', window.chatSettings);
  }, [userData]);

  useEffect(() => {
    // Load Genesys Cloud bootstrap and widget scripts
    const bootstrapUrl = window.chatSettings?.bootstrapUrl || '';
    const widgetUrl = window.chatSettings?.widgetUrl || '';
    if (!bootstrapUrl) return;
    const bootstrapScript = document.createElement('script');
    bootstrapScript.src = bootstrapUrl;
    bootstrapScript.async = true;
    bootstrapScript.onload = () => {
      console.debug('[CloudChatWrapper] Genesys bootstrap loaded');
      if (!widgetUrl) return;
      const widgetScript = document.createElement('script');
      widgetScript.src = widgetUrl;
      widgetScript.async = true;
      widgetScript.onload = () => {
        console.debug('[CloudChatWrapper] Genesys widgets loaded');
      };
      widgetScript.onerror = (e) => {
        console.error('[CloudChatWrapper] Failed to load widgets script', e);
      };
      document.body.appendChild(widgetScript);
    };
    bootstrapScript.onerror = (e) => {
      console.error('[CloudChatWrapper] Failed to load bootstrap script', e);
    };
    document.body.appendChild(bootstrapScript);

    // Provide openGenesysChat logic
    window.openGenesysChat = () => {
      if (window.Genesys) {
        try {
          window.Genesys('command', 'Messenger.open');
        } catch (e) {
          console.error(
            '[CloudChatWrapper] Error opening Genesys Messenger',
            e,
          );
        }
      }
    };

    return () => {
      // Cleanup scripts and chatSettings
      document.body.removeChild(bootstrapScript);
      if (widgetUrl) {
        const widgetScript = document.querySelector(
          'script[src="' + widgetUrl + '"]',
        );
        if (widgetScript) document.body.removeChild(widgetScript);
      }
      if (window.chatSettings) delete window.chatSettings;
      if (window.openGenesysChat) delete window.openGenesysChat;
      console.debug('[CloudChatWrapper] Cleanup complete');
    };
  }, [userData]);

  return null;
}
