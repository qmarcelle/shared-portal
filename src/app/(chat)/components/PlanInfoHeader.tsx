'use client';
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { Title } from '@/components/foundation/Title';

/**
 * Show current plan when multiple plans exist
 */
export default function PlanInfoHeader() {
  const { formInputs } = useChatStore();
  if (formInputs.length <= 1) return null;
  const plan = formInputs.find((f) => f.id === 'PLAN_ID');

  return (
    <Title
      className="plan-info-header"
      text={`You are chatting about plan ${plan?.value || ''}`}
    />
  );
}
