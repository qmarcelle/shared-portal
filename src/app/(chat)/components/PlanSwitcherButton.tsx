'use client';
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { Button } from '@/components/foundation/Button';
import { usePlanStore } from '../../../userManagement/stores/planStore';

/**
 * "Switch Plan" inside pre-chat window
 */
export default function PlanSwitcherButton() {
  const { endChat } = useChatStore();
  const { plans, selectedPlanId, openPlanSwitcher } = usePlanStore();

  if (plans.length <= 1) return null;

  const handleSwitch = () => {
    endChat();
    openPlanSwitcher();
  };

  return (
    <Button
      label="Switch Plan"
      callback={handleSwitch}
      className="plan-switch-button"
      type="secondary"
    />
  );
}
