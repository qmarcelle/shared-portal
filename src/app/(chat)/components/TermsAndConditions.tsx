'use client';
// import TCs from '@/app/@chat/config/terms'; // { default: '…', lobA: '…', … }
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { RichText } from '@/components/foundation/RichText';

/**
 * Render T&C based on LOB
 */
export default function TermsAndConditions() {
  const { userData } = useChatStore();
  // const text = TCs[userData.LOB] || TCs.default;
  return (
    <RichText
      spans={['Terms and Conditions']}
      className="terms-and-conditions"
      type="body-2"
    />
  );
}
