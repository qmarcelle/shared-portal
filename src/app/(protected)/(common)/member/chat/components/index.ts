/**
 * Component Exports
 * Centralizes all component exports following the new consolidated structure
 */

// Main Components
export { ChatTrigger } from './ChatTrigger';
export { ChatWidget } from './ChatWidget';
export { PlanInfoHeader } from './PlanInfoHeader';

// Error Handling
export { ChatErrorBoundary } from './shared/ChatErrorBoundary';

// Integration Components (to be consolidated)
export { GenesysScripts } from './GenesysScripts';
