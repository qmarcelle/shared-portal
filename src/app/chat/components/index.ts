// Core Components
export * from './core/ChatButton';
export * from './core/ChatErrorBoundary';
export * from './core/ChatHeader';
export * from './core/ChatWidget';

// Business Hours Components
export * from './business-hours/BusinessHoursNotification';

// Eligibility Components
export * from './eligibility/EligibilityCheck';

// Plan Switcher Components will be added here when implemented
// export * from './plan-switcher/PlanSwitcher';
