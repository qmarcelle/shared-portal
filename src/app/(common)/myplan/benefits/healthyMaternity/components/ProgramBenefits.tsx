/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable react/jsx-key */
import { Column } from '@/components/foundation/Column';
import { Header } from '@/components/foundation/Header';
import { Row } from '@/components/foundation/Row';
import { Spacer } from '@/components/foundation/Spacer';
import { IComponent } from '../../../components/IComponent';
import { TextBox } from '../../../components/foundation/TextBox';
import { BenefitDetails } from '../models/benefit_details';

interface ProgramBenefitsProps extends IComponent {
  benefits: BenefitDetails[];
}

export const ProgramBenefits = ({ benefits }: ProgramBenefitsProps) => {
  return (
    <>
      <Spacer size={12} />
      <Header className="title-2 px-3" text="Program Benefits"></Header>
      <Spacer size={25} />
      <Row className="hidden sm:flex px-3">
        {benefits.slice(0, benefits.length).map((item, index) => (
          <Column className="md:pr-4 md:w-1/3">
            {item.benefitIcon}
            <Spacer size={5} />
            <TextBox text={item.benefitLabel} className="font-bold" />
            <Spacer size={5} />
            <TextBox text={item.benefitCopy} />
          </Column>
        ))}
      </Row>
      <Column className="block sm:hidden px-3">
        {benefits.slice(0, benefits.length).map((item, index) => (
          <Column className="pr-4 mb-4">
            {item.benefitIcon}
            <Spacer size={5} />
            <TextBox text={item.benefitLabel} className="font-bold" />
            <Spacer size={5} />
            <TextBox text={item.benefitCopy} />
          </Column>
        ))}
      </Column>
    </>
  );
};
