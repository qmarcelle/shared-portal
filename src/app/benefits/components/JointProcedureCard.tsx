import { Card } from '@/components/foundation/Card';
import { Column } from '@/components/foundation/Column';
import { RichText } from '@/components/foundation/RichText';
import { Spacer } from '@/components/foundation/Spacer';
import { IComponent } from '@/components/IComponent';

interface JointProcedureCardProps extends IComponent {
  phoneNumber: string;
}

export const JointProcedureCard = ({
  className,
  phoneNumber,
}: JointProcedureCardProps) => {
  return (
    <Card className={className} type="info">
      <Column>
        <h2 className="title-3">Call Before Scheduling Your Joint Procedure</h2>
        <Spacer size={32} />
        <RichText
          type="body-1"
          spans={[
            <span key={0}>
              {
                'Your plan requires giving us a call before pursuing knee, hip, or spine procedures. Give us a call at '
              }
            </span>,
            <span key={1}>{`[${phoneNumber}] or `}</span>,
            <span className="link-white-text" key={2}>
              <a>start a chat</a>
            </span>,
            <span key={3}>{' now.'}</span>,
          ]}
        />
      </Column>
    </Card>
  );
};
