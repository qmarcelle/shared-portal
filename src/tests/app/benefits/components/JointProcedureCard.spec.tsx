import { JointProcedureCard } from '@/app/benefits/components/JointProcedureCard';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

const renderUI = () => {
  return render(
    <JointProcedureCard
      className="large-section px-0 m-0 text-white"
      phoneNumber={'1800-000-000'}
    />,
  );
};

describe('JointProcedureCard Section', () => {
  it('should render the UI correctly', async () => {
    const { container } = renderUI();

    expect(
      screen.getByText('Call Before Scheduling Your Joint Procedure'),
    ).toBeVisible();
    expect(
      screen.getByText(
        'Your plan requires giving us a call before pursuing knee, hip, or spine procedures. Give us a call at',
      ),
    ).toBeVisible();
    expect(screen.getByText('[1800-000-000] or')).toBeVisible();
    expect(container).toMatchSnapshot();
  });
});
