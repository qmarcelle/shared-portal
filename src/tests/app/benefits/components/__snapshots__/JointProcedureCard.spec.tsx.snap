// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JointProcedureCard Section should render the UI correctly 1`] = `
<div>
  <div
    class="card-info large-section px-0 m-0 text-white"
  >
    <div
      class="flex flex-col"
    >
      <h2
        class="title-3"
      >
        Call Before Scheduling Your Joint Procedure
      </h2>
      <span
        style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
      />
      <span
        class="rich-text body-1 undefined"
      >
        <span>
          Your plan requires giving us a call before pursuing knee, hip, or spine procedures. Give us a call at 
        </span>
        <span>
          [1800-000-000] or 
        </span>
        <span
          class="link-white-text"
        >
          <a>
            start a chat
          </a>
        </span>
        <span>
           now.
        </span>
      </span>
    </div>
  </div>
</div>
`;
